include/adaflo/sharp_interface_util.h

```diff
@@ -0,0 +1,1108 @@
+                           const DoFHandler<spacedim, spacedim> &background_dofhandler,
+                           const Mapping<spacedim, spacedim> &   background_mapping,
+                           const VectorType &                    velocity_vector,
+                           const DoFHandler<dim, spacedim> &     euler_dofhandler,
+                           const Mapping<dim, spacedim> &        euler_mapping,
+                           VectorType &                          euler_coordinates_vector)
+    {
+      FEValues<dim, spacedim> fe_eval(
+        euler_mapping,
+        euler_dofhandler.get_fe(),
+        Quadrature<dim>(euler_dofhandler.get_fe().get_unit_support_points()),
+        update_quadrature_points);
+
+      Vector<double>                       temp;
+      std::vector<types::global_dof_index> temp_dof_indices;
+
+      auto euler_coordinates_vector_temp = euler_coordinates_vector;
+      auto euler_coordinates_vector_bool = euler_coordinates_vector;
+      euler_coordinates_vector_bool      = 0.0;
+
+      const std::vector<bool>                    marked_vertices;
+      const GridTools::Cache<spacedim, spacedim> cache(
+        background_dofhandler.get_triangulation(), background_mapping);
+      const double tolerance = 1e-10;
+      auto         cell_hint = background_dofhandler.get_triangulation().begin_active();
+
+      for (const auto &cell : euler_dofhandler.active_cell_iterators())
+        {
+          fe_eval.reinit(cell);
+
+          temp.reinit(fe_eval.dofs_per_cell);
+          temp_dof_indices.resize(fe_eval.dofs_per_cell);
+
+          cell->get_dof_indices(temp_dof_indices);
+          cell->get_dof_values(euler_coordinates_vector, temp);
+
+          for (const auto q : fe_eval.quadrature_point_indices())
+            {
+              // if (euler_coordinates_vector_bool[temp_dof_indices[q]] == 1.0)
+              //  continue;
+
+              euler_coordinates_vector_bool[temp_dof_indices[q]] = 1.0;
+
+              const auto velocity = [&] {
+                const auto cell_and_reference_coordinate =
+                  GridTools::find_active_cell_around_point(cache,
+                                                           fe_eval.quadrature_point(q),
+                                                           cell_hint,
+                                                           marked_vertices,
+                                                           tolerance);
+
+                std::vector<double> buffer(
+                  background_dofhandler.get_fe().n_dofs_per_cell());
+
+                typename DoFHandler<spacedim, spacedim>::active_cell_iterator
+                  background_cell{&background_dofhandler.get_triangulation(),
+                                  cell_and_reference_coordinate.first->level(),
+                                  cell_and_reference_coordinate.first->index(),
+                                  &background_dofhandler};
```

Comment:
Is there a reason why you are not simply using the constructor,
```suggestion
                  background_cell(&background_dofhandler.get_triangulation(),
                                  cell_and_reference_coordinate.first->level(),
                                  cell_and_reference_coordinate.first->index(),
                                  &background_dofhandler);
```